SYCL/OneapiDeviceSelector/case_sensitivity.cpp

```diff
@@ -0,0 +1,8 @@
+
+// does not actually require OpenCL or GPU. Just testing parsing.
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %S/Inputs/trivial.cpp -o %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="OPENCL:*" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="opencl:*" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:GPU" %t.out
+// RUN: env ONEAPI_DEVICE_SELECTOR="*:gpu" %t.out
```

Comment:
Would it make sense to have a test where it mixes upper and lowercase letters? I.e. something like `OPeNcL` or even an actual common spelling like `OpenCL`.

Reply:
Maybe. What if I removed all these cases for just one: "OpEnCl:GpU" ?